File: frontend/types/activity.ts
export interface ActivityLog {
  id: number;
  user_id: number | null;
  action: string;
  resource_type: string | null;
  resource_id: number | null;
  organization_id: number | null;
  details: Record<string, any> | null;
  ip_address: string | null;
  user_agent: string | null;
  created_at: string; // ISO datetime string
  user_name?: string | null;
  organization_name?: string | null;
}

export interface ActivityLogList {
  logs: ActivityLog[];
  total: number;
  page: number;
  page_size: number;
}

export interface ActivityLogFilters {
  page?: number;
  page_size?: number;
  user_id?: number;
  action?: string;
  resource_type?: string;
  organization_id?: number;
}

File: backend/app/schema/activity.py
from pydantic import BaseModel
from typing import Optional, Dict, Any, List
from datetime import datetime


class ActivityLogOut(BaseModel):
    """Schema for activity log output."""
    id: int
    user_id: Optional[int] = None
    action: str
    resource_type: Optional[str] = None
    resource_id: Optional[int] = None
    organization_id: Optional[int] = None
    details: Optional[Dict[str, Any]] = None
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None
    created_at: datetime
    
    # Optional user/organization info
    user_name: Optional[str] = None
    organization_name: Optional[str] = None
    
    class Config:
        from_attributes = True


class ActivityLogList(BaseModel):
    """Schema for paginated activity log list."""
    logs: List[ActivityLogOut]
    total: int
    page: int
    page_size: int

